ADR/0028-component-dependencies.md

```diff
@@ -0,0 +1,179 @@
+# 28. Component Dependencies
+
+* Date 2023-08-15
+
+## Status
+
+Accepted
+
+## Context
+
+As an AppStudio user, I want to be able to build and test multiple coupled components which depend on each other by digest reference. I want that process to be easy.
+
+There are three use cases in scope for this document:
+
+* A user team has their own **common parent image**. When they propose an update to their base image with new content, they want to see if that’s going to break any of their Components that depend on it before merging. AppStudio should posit what rebuilds of those Components will look like and if they will pass their tests, and report that feedback back to the original pull request that updated content of the common parent image ([RHTAP-967](https://issues.redhat.com/browse/RHTAP-967)). A real example of this is in stolostron, where a [layered image component](https://github.com/stolostron/console/blob/main/Dockerfile.mce.prow#L20) refers to a [common parent image](https://github.com/stolostron/common-nodejs-parent).
+  * In this case, the dependent images must be rebuilt to include the common parent image update in order to determine the actual effect of the proposed change.
+  * This is a many-to-one dependency. Many component images depend on one common parent image.
```

Comment:
Is it *alway* a many-to-one ? Even though one "final" image has only one base image, if the `Dockerfile` used is multi-stage, there is usually different base image for different stages, and one stages that is used to generate could fail if that base image is updated.